macq/generate/pddl/generator.py

```diff
@@ -401,16 +402,27 @@ def generate_plan(self, from_ipc_file: bool = False, filename: str = None):
                     "problem": open(self.pddl_prob, "r").read(),
                 }
 
+                headers = {"persistent": "true"}
+
                 def get_api_response(delays: List[int]):
                     if delays:
                         sleep(delays[0])
                         try:
-                            resp = requests.post(
-                                "http://solver.planning.domains/solve",
-                                verify=False,
-                                json=data,
-                            ).json()
-                            return [act["name"] for act in resp["result"]["plan"]]
+                            service_url = "https://solver.planning.domains:5001/package/lama-first/solve"
+                            solve_request = requests.post(service_url, json=data, headers=headers).json()
+                            celery_result = requests.get("https://solver.planning.domains:5001/" +
+                                                         solve_request['result'])
+                            print('Computing...')
+                            while celery_result.json().get("status", "") == 'PENDING':
+                                sleep(delays[0])
+                                celery_result = requests.get("https://solver.planning.domains:5001/" +
+                                                             solve_request['result'])
+                            sas_plan = celery_result.json()['result']['output']['sas_plan']
+                            actions_with_objects = re.findall(r'\((.*?)\)', sas_plan)
```

Comment:
We guaranteed that this will return them in the sorted order? Worried it might scramble the sequence of actions, but I could be wrong there.

Reply:
re.findall findes strings by order they accured so it should kep the order they had in the output of the solver.